### PR TITLE
Preserve asset cache between loads

### PR DESCRIPTION
**Pull request by stevenvergenz**
* Add `GLTFSceneImporter.LoadMaterialAsync` API
* No longer flush the asset cache between LoadScene/Node/Material/Texture calls
* Add timeout to WebRequestLoader
* Fix exception in SimpleWebServer
* Make importer Construct functions asynchronous if they might load new data (images, buffers)

### CODE REVIEW

**Comment by stevenvergenz** [comment]
Do not merge yet, still stabilizing.

**Comment by stevenvergenz** [comment]
@blgrossMS Yes it's ready for a review.

File: UnityGLTF/Assets/UnityGLTF/Examples/SimpleWebServer.cs [hunk 1]
```diff
@@ -187,13 +187,14 @@ private void Process(HttpListenerContext context)
...
-
 			if (File.Exists(filename))
 			{
+				context.Response.StatusCode = (int)HttpStatusCode.OK;
+
 				try
 				{
-					Stream input = new FileStream(filename, FileMode.Open);
-
+					Stream input = File.OpenRead(filename); //new FileStream(filename, FileMode.Open);
```
Comment (blgrossMS) [inline UnityGLTF/Assets/UnityGLTF/Examples/SimpleWebServer.cs]: remove comment

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 2]
```diff
@@ -212,11 +216,6 @@ public IEnumerator LoadScene(int sceneIndex = -1, bool showSceneObj = true, Acti
 		/// <returns></returns>
 		public async Task LoadNodeAsync(int nodeIndex)
 		{
-			if (_gltfRoot == null)
```
Comment (blgrossMS) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Why was this removed?

File: UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs [hunk 3]
```diff
@@ -180,9 +180,13 @@ public async Task LoadSceneAsync(int sceneIndex = -1, bool showSceneObj = true,
 				{
 					await LoadJson(_gltfFileName);
 				}
-				await _LoadScene(sceneIndex, showSceneObj);
 
-				Cleanup();
+				if (_assetCache == null)
+				{
+					_assetCache = new AssetCache(_gltfRoot);
```
Comment (blgrossMS) [inline UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: did you ensure with this change the asset cache still gets cleaned up properly?

Comment (stevenvergenz) [reply UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs]: Yes and no. The asset cache persists indefinitely during the lifetime of the importer, then is disposed when the importer is disposed. Since you can't load a second glTF document from a single importer anyway, it works out.
